h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -504,29 +509,31 @@ public static void bulkBuildModels(String modelType, Job job, ModelBuilder<?, ?,
     }
     for( int i=0; i<N; ++i ) //all sub-models must be completed before the main model can be built
       try {
-        submodel_tasks[i].join();
+        final H2O.H2OCountedCompleter task = submodel_tasks[i];
+        if (task != null) task.join();
       } catch(RuntimeException t){
-        if(rt == null) rt = t;
+        if (rt == null) rt = t;
       }
     if(rt != null) throw rt;
   }
 
-  private void buildMainModel() {
```

Comment:
Please, the changes minimal when changing core classes.

Reply:
I know but harder to review - it is really easy to introduce a bug this way

Reply:
closed